nix/copySource.nix

```diff
@@ -23,35 +23,40 @@ let
       mainFileRegex = "(.*/)?" + recipeAttrs.pname + "\.el";
       isMainFile = file: match mainFileRegex file != null;
     in
-      rec {
-        inherit (recipeAttrs) pname;
-        packageFiles =
-          filter (file: match "(.*/)?flycheck_.+\.el" file == null)
-            (expandPackageFiles srcRoot recipeAttrs.files);
-        sourceFiles = filter isElisp packageFiles;
-        mainFile =
-          if length sourceFiles == 1
-          then head sourceFiles
-          else head (filter isMainFile sourceFiles);
-      };
+    rec {
+      inherit (recipeAttrs) pname;
+      packageFiles =
+        filter (file: match "(.*/)?flycheck_.+\.el" file == null)
+          (expandPackageFiles srcRoot recipeAttrs.files);
+      sourceFiles = filter isElisp packageFiles;
+      mainFile =
```

Comment:
Because Nix is a lazy language, you can put the definition of `mainFiles` above this line.